service/collector/aws.go

```diff
@@ -24,9 +26,21 @@ func (c Collector) getARNs() ([]string, error) {
 	arnsMap := make(map[string]bool)
 	for _, awsConfig := range awsConfigs.Items {
 		arn, err := credential.GetARN(c.k8sClient, &awsConfig)
-		if err != nil {
+		// Collect as many ARNs as possible in order to provide most metrics.
+		// Ignore old cluster which do not have credential.
+		if credential.IsCredentialNameEmptyError(err) ||
+			credential.IsCredentialNamespaceEmptyError(err) {
+			continue
```

Comment:
I would prefer to have one clause per line. This weird long conditions do we also have in the `azure-operator` and I cannot really parse that. Not a blocker though.

Reply:
Would a `switch` help here?
```
		switch {
		case credential.IsCredentialNameEmptyError(err):
			continue
		case credential.IsCredentialNamespaceEmptyError(err):
			continue
		case err != nil:
			return nil, microerror.Mask(err)
		}
```

Reply:
No blocker on using switch but from `fmt` I'd say single conditions with an `else if` is our preferred style.

https://github.com/giantswarm/fmt/blob/master/go/errors.md#matching-errors

Reply:
IMO the usual `fmt` structure should be applied. It is all not new. Just apply what we have already.